File: hw2/src/plane.cpp
#include "plane.h"

Plane::Plane(int speed, float distance, int maxDistance, int carryingCapacity)
    : BaseTransport(speed, distance), m_maxDistance(maxDistance), m_carryingCapacity(carryingCapacity)
{
}

void Plane::Out(std::ostream& out)
{
    BaseTransport::Out(out);
    out << ", max distance = " << m_maxDistance << ", carrying capacity = " << m_carryingCapacity << std::endl;
}

std::string_view Plane::GetName()
{
    return "Plane";
}

File: hw3/src/plane.py
'''
plane.py - содержит описание самолета
'''

from base_transport import *

# Самолет
class Plane(BaseTransport):
    # Инициализация
    def __init__(self, speed, distance, max_distance, carrying_capacity):
        super().__init__(speed, distance)
        self.max_distance = max_distance
        self.carrying_capacity = carrying_capacity

    # Конвертация в строку
    def __str__(self):
        return super().__str__() + f', max_distance = {self.max_distance}, carrying_capacity = {self.carrying_capacity}'
